### ISSUE TITLE
Questions about training scripts

### ISSUE THREAD

**Issue by passing2961** [test-time-training/discover#7]
Hello,

I have two questions: 
1. Would you be able to share the scripts for the Kernel, Algorithm, and Biology tasks that are presented in the main table of Figure 1? Do they share all same configurations?
2. In Table 2, it seems that you ran OpenEvolve with GPT-OSS-120B. Could you share the detailed configuration you used—for example, the max iterations, temperature, max_tokens, number of islands, and any other relevant settings?

**Comment by passing2961** [comment]
@mertyg Thank you for the updated codes. However, I couldn't find training scripts for each task (e.g., `scripts/thinker/ac1.sh`).

**Comment by vinid** [comment]
for openevolve:

configuration is the default configuration provided in openevolve for math tasks (4 islands, 60 population size...). We use the same initial program provided by openevolve as a starting point.

The only two very relevant differences are `max_iterations: 25600` and `parallel_evaluations: 512`. 

In addition to the above, note that we had to implement cpu pinning and assign two CPUs to each parallel evaluation (briefly mentioned this in #6).

**Comment by passing2961** [comment]
@koceja I see. By the way, I trained my custom model on erdos problem using your code for 25 epochs. And I got the following score at last epoch:

```
"env/all/raw_score": 0.3866294036405717, "env/all/raw_score/min": 0.3809771821788439, "env/all/raw_score/max": 0.5045815374252317, "env/all/initial_raw_score": -0.38133746868803853,
```
For final performance when you reported in your paper, did you use "raw_score" or "raw_score/min"?

**Comment by vinid** [comment]
i'd recommend running extracting the best construction from the samper states and passing it through the verifier (e.g., you can use [alphaevolve's](https://colab.research.google.com/github/google-deepmind/alphaevolve_results/blob/master/mathematical_results.ipynb) verifiers).

eventually what matters in this specific task is if you have found a new upper bound (and you can easily compare with previous numbers). Also we have some additional code [here](https://github.com/test-time-training/discover/tree/main/results/mathematics) for plotting

**Comment by koceja** [comment]
Yes, it should be `env/all/raw_score/min`. I realized that our documentation is very lacking about retrieving the final performance so I updated the docs [here](https://github.com/test-time-training/discover/blob/main/docs/reproducing.md#getting-final-performance). Thank you for pointing this out, let me know if the updated docs make more sense.

**Comment by passing2961** [comment]
@vinid @koceja Thank you very much for the incredibly fast response and for improving the documentation. It has become much clearer now.

While running the Erdős problem, I was able to view the actual generated codes in the wandb history. Of course, I understand that I can export them from wandb if needed. However, I have a minor question:

- If I want to make sure that all actually generated results are saved locally, should I use wandb in offline mode? Or are the `constructions` and `code` (key value) already stored in the existing `tinker_logs` folder, specifically in the `puct_sampler_* files`?

Just wanted to clarify which approach is correct.

**Comment by vinid** [comment]
> I was able to view the actual generated codes in the wandb history.

That's correct! Just keep in mind that for the Erdős problem, the actual construction is what really matters. Even if you use the same seed, running the code again might not give you the same construction. Your best source of truth will be the `puct_sampler_*` logs. Not **constructions and codes** are stored there, only the best ones. 

A side note: In my experiments, wandb is always lagging behind actual training (meaning that training might be at step 10 but wandb logs only show step 7, I assume this is due to the async upload). So remember to check that your data is on wandb before killing the run.
